### PR TITLE
MNT: Restore auto-adding Axes3D to their parent figure on init

### PR DESCRIPTION
**Pull request by tacaswell**
Only do this when directly created by the user, not via our machinery.

## PR Summary

Alternative to #19413.

This still needs tests and updates to the docs.

## PR Checklist

- [x] Has pytest style unit tests (and `pytest` passes).
- [x] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (run `flake8` on changed files to check).
- [ ] New features are documented, with examples if plot related.
- [x] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [x] Conforms to Matplotlib style conventions (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).
- [x] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [ ] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).

### CODE REVIEW

**Comment by jklymak** [comment]
This passes the labelled test if you go back to the old way of calling:

```python
@mpl3d_image_comparison(['axes3d_labelpad.png'], remove_text=False)
def test_axes3d_labelpad():
    fig = plt.figure()
    ax = Axes3D(fig)
```

OTOH, I was hoping we would deprecate calling Axes3D this way because it really is illogical to have it behave differently from every other type of axes.

**Comment by tacaswell** [comment]
I ended up going with @jklymak 's approach of a public facing kwarg (to give users a migration path), but did it via kwarg-popping so it does not show up in the signature.

I think this is a lighter touch than #19413.

File: doc/api/next_api_changes/behavior/18564-AL.rst [hunk 1]
```diff
@@ -1,8 +1,22 @@
...
+In the future, `.Axes3D` will need to be explicitly added to the
+figure ::
+
+  fig = Figure()
+  # create Axes3D
+  ax = Axes3d(fig)
+  # add to Figure
+  fig.add_axes(ax)
+
+as needs to be done for other `.axes.Axes` sub-classes.
```
Comment (jklymak) [inline doc/api/next_api_changes/behavior/18564-AL.rst:22]:
```suggestion
as needs to be done for other `.axes.Axes` sub-classes. Or, a 3-d 
projection can be made via::

    fig.add_subplot(projection='3d')
```

Comment (jklymak) [reply doc/api/next_api_changes/behavior/18564-AL.rst:22]: .. otherwise this all seems good to me.

File: lib/mpl_toolkits/mplot3d/axes3d.py [hunk 2]
```diff
@@ -64,6 +64,15 @@ def __init__(
             Other axes to share z-limits with.
         proj_type : {'persp', 'ortho'}
             The projection type, default 'persp'.
+        auto_add_to_figure : bool, default True
```
Comment (QuLogic) [inline lib/mpl_toolkits/mplot3d/axes3d.py:67]:
```suggestion
        auto_add_to_figure : bool, default: True
```

File: lib/mpl_toolkits/mplot3d/axes3d.py [hunk 3]
```diff
@@ -96,10 +105,11 @@ def __init__(
         if sharez is not None:
             self._shared_z_axes.join(self, sharez)
             self._adjustable = 'datalim'
-
+        auto_add_to_figure = kwargs.pop('auto_add_to_figure', True)
         super().__init__(
             fig, rect, frameon=True, box_aspect=box_aspect, *args, **kwargs
         )
+
```
Comment (QuLogic) [inline lib/mpl_toolkits/mplot3d/axes3d.py:112]: Not sure the blank lines should move around this way.

**Review by QuLogic — APPROVED** [review]

**Comment by tacaswell** [comment]
I think I fixed the doc build (by updating a very old whats new example!).

**Review by jklymak — APPROVED** [review]

**Pull request merged**